src/ray/gcs/gcs_client/gcs_client.h

```diff
@@ -215,10 +216,32 @@ class RAY_EXPORT GcsClient : public std::enable_shared_from_this<GcsClient> {
   /// service failure is detected.
   ///
   /// \param type The type of GCS service failure.
-  void GcsServiceFailureDetected(rpc::GcsServiceFailureType type);
+  /// \param failure_handled_callback The callback function to be called after reconnected
+  /// to gcs successfully. \return Returns true if the reconnect request is accepted,
+  /// false if the client is in reconnecting process and the reconnect request is
+  /// rejected.
+  bool GcsServiceFailureDetected(
```

Comment:
Can we remove this function? This seems to be just unnecessary wrapper?

Reply:
Nice. It is legacy code, and in terms of current implementation, we can remove it. Meanwhile, after removal `rpc::GcsServiceFailureType::RPC_DISCONNECT` and `rpc::GcsServiceFailureType::GCS_SERVER_RESTART` turn to useless and can be removed.